### PR TITLE
Inject the BuildInfoService into the NewSessionPresenter and SavedSes…

### PR DESCRIPTION
**Pull request by DenverM80**
invalid_endpoint
…sionProvider, require in SavedSessionStore and CreateConnectionTask

### CODE REVIEW

**Comment by DenverM80** [comment]
When I tested, the error just said "Something went wrong", and then found that the buildVersion included in the change for the user-agent assumed that buildVersion is always available, when its only actually available in production builds.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/CreateConnectionTask.java [hunk 1]
```diff
@@ -67,15 +71,18 @@ public static Session createConnection(final NewSessionModel newSessionModel, fi
                     LOG.error("Invalid Access ID or Secret Key: ", e);
                     alert.showAlert(resourceBundle.getString("invalidIDKEYMessage"));
                 }
+            } else if (e.getStatusCode() == 301) {
+                LOG.error("BlackPearl return an unexpected status code, indicating we are attempting to make a data path request on the Management port: ", e);
+                alert.showAlert(resourceBundle.getString("invalidEndpointMessage"));
             } else {
-                LOG.error("BlackPearl return an unexpected status code we did not expect: {}", e);
+                LOG.error("BlackPearl return an unexpected status code: ", e);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/CreateConnectionTask.java]: `return` should be `returned`

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/CreateConnectionTask.java [hunk 2]
```diff
@@ -67,15 +71,18 @@ public static Session createConnection(final NewSessionModel newSessionModel, fi
...
                 alert.showAlert(resourceBundle.getString("unexpectedStatusMessage"));
             }
         } catch (final IOException ioe) {
             LOG.error("Encountered a networking error: ", ioe);
             alert.showAlert(resourceBundle.getString("networkErrorMessage"));
         } catch (final RuntimeException rte) {
-            LOG.error("Something went wrong.", rte);
+            LOG.error("Something went wrong: ", rte);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/CreateConnectionTask.java]: remove the punctuation in the log message.  The logger already has some when it prints out the message

File: dsb-gui/src/main/resources/lang_en.properties [hunk 3]
```diff
@@ -109,7 +109,7 @@ failToAuthenticateMessage=Failed To authenticate session : Client's clock is not
 invalidIDKEY=Invalid ID and KEY
 invalidIDKEYMessage=Invalid Access ID or Secret Key
 unexpectedStatus=Unexpected Status
-unexpectedStatusMessage=BlackPearl return an unexpected status code we did not expect
+unexpectedStatusMessage=BlackPearl return an unexpected status code
```
Comment (rpmoore) [inline dsb-gui/src/main/resources/lang_en.properties]: `return` should be `returned`

**Comment by DenverM80** [comment]
updated.

**Pull request merged**